src/java.desktop/share/classes/javax/swing/text/html/CSS.java

```diff
@@ -2423,6 +2471,16 @@ Object toStyleConstants(StyleConstants key, View v) {
             return c;
         }
 
+        @Override
+        public int hashCode() {
+            return (c != null) ? c.hashCode() : 0;
+        }
+
+        @Override
+        public boolean equals(Object val) {
+            return val instanceof CSS.ColorValue color && c.equals(color.c);
```

Comment:
Is it possible that the value of `c` is null? You have the null-check in `hashCode` but you don't test it in `equals`.

`ColorValue.parseCssValue` returns `null` if it can't parse the color. So, `c` should never be `null`.